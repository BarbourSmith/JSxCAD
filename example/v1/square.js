import { square, writePdf } from '@jsxcad/api-v1';

writePdf({ path: '/tmp/square.pdf' }, square(30));
