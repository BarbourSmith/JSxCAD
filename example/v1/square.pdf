%PDF-1.5
1 0 obj << /Pages 2 0 R /Type /Catalog >> endobj
2 0 obj << /Count 1 /Kids [ 3 0 R ] /Type /Pages >> endobj
3 0 obj <<
  /Contents 4 0 R
  /MediaBox [ 0 0 595.275590176 841.889763249 ]
  /Parent 2 0 R
  /Type /Page
>>
endobj
4 0 obj << >>
stream
0.096000000 w
0.000000000 0.000000000 0.000000000 rg
0.000000000 841.889763249 m
85.039370025 756.850393224 l
85.039370025 841.889763249 l
h
f
85.039370025 756.850393224 m
0.000000000 841.889763249 l
0.000000000 756.850393224 l
h
f
endstream
endobj
trailer << /Root 1 0 R /Size 4 >>
%%EOF