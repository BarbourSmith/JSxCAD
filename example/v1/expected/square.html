<html>
  <head>
    <title>JSxCAD Viewer</title>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0"
    />
  </head>
  <body>
    <script type="module">
      import { toDomElement } from 'https://gitcdn.link/cdn/jsxcad/JSxCAD/master/es6/jsxcad-ui-notebook.js';

      const notebook = [
        {
          view: {
            width: 512,
            height: 256,
            position: [100, -100, 100],
            geometry: {
              type: 'layout',
              layout: {
                size: [32.00000000000001, 32.00000000000001],
                margin: 1,
                title: '32 x 32 : ',
              },
              marks: [
                [0, 0, 0],
                [32.00000000000001, 32.00000000000001, 0],
              ],
              content: [
                {
                  type: 'item',
                  content: [
                    {
                      type: 'disjointAssembly',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'z0Surface',
                              z0Surface: [
                                [
                                  [15.000000000000004, 15, 0],
                                  [-14.999999999999998, 15.000000000000007, 0],
                                  [-15.000000000000004, -14.999999999999998, 0],
                                  [14.999999999999996, -15.000000000000005, 0],
                                ],
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                {
                  type: 'disjointAssembly',
                  content: [
                    {
                      type: 'disjointAssembly',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'paths',
                              tags: ['compose/non-positive', 'color/red'],
                              paths: [
                                [
                                  [16.000000000000004, 16.000000000000004, 0],
                                  [-16.000000000000004, 16.000000000000004, 0],
                                  [-16.000000000000007, -16, 0],
                                  [16, -16.000000000000007, 0],
                                ],
                              ],
                            },
                          ],
                        },
                        {
                          type: 'paths',
                          tags: ['compose/non-positive', 'color/red'],
                          paths: [
                            [
                              null,
                              [-15.642857142857146, 18.500000000000004, 0],
                              [-14.857142857142861, 18.500000000000004, 0],
                              [-15.285714285714288, 17.92857142857143, 0],
                              [-15.071428571428575, 17.92857142857143, 0],
                              [-14.928571428571432, 17.85714285714286, 0],
                              [-14.857142857142861, 17.78571428571429, 0],
                              [-14.785714285714288, 17.571428571428577, 0],
                              [-14.785714285714288, 17.42857142857143, 0],
                              [-14.857142857142861, 17.21428571428572, 0],
                              [-15.000000000000004, 17.071428571428577, 0],
                              [-15.214285714285717, 17.000000000000004, 0],
                              [-15.428571428571432, 17.000000000000004, 0],
                              [-15.642857142857146, 17.071428571428577, 0],
                              [-15.714285714285717, 17.142857142857146, 0],
                              [-15.78571428571429, 17.28571428571429, 0],
                            ],
                            [null],
                            [
                              null,
                              [-14.285714285714288, 18.142857142857146, 0],
                              [-14.285714285714288, 18.21428571428572, 0],
                              [-14.214285714285717, 18.35714285714286, 0],
                              [-14.142857142857146, 18.42857142857143, 0],
                              [-14.000000000000004, 18.500000000000004, 0],
                              [-13.714285714285717, 18.500000000000004, 0],
                              [-13.571428571428575, 18.42857142857143, 0],
                              [-13.500000000000004, 18.35714285714286, 0],
                              [-13.428571428571432, 18.21428571428572, 0],
                              [-13.428571428571432, 18.071428571428577, 0],
                              [-13.500000000000004, 17.92857142857143, 0],
                              [-13.642857142857146, 17.71428571428572, 0],
                              [-14.357142857142861, 17.000000000000004, 0],
                              [-13.357142857142861, 17.000000000000004, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-11.785714285714288, 18.000000000000004, 0],
                              [-11.000000000000004, 17.000000000000004, 0],
                            ],
                            [
                              null,
                              [-11.000000000000004, 18.000000000000004, 0],
                              [-11.785714285714288, 17.000000000000004, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-9.285714285714288, 18.500000000000004, 0],
                              [-8.500000000000004, 18.500000000000004, 0],
                              [-8.92857142857143, 17.92857142857143, 0],
                              [-8.714285714285717, 17.92857142857143, 0],
                              [-8.571428571428573, 17.85714285714286, 0],
                              [-8.500000000000004, 17.78571428571429, 0],
                              [-8.42857142857143, 17.571428571428577, 0],
                              [-8.42857142857143, 17.42857142857143, 0],
                              [-8.500000000000004, 17.21428571428572, 0],
                              [-8.642857142857146, 17.071428571428577, 0],
                              [-8.85714285714286, 17.000000000000004, 0],
                              [-9.071428571428573, 17.000000000000004, 0],
                              [-9.285714285714288, 17.071428571428577, 0],
                              [-9.35714285714286, 17.142857142857146, 0],
                              [-9.42857142857143, 17.28571428571429, 0],
                            ],
                            [null],
                            [
                              null,
                              [-7.928571428571431, 18.142857142857146, 0],
                              [-7.928571428571431, 18.21428571428572, 0],
                              [-7.857142857142859, 18.35714285714286, 0],
                              [-7.785714285714288, 18.42857142857143, 0],
                              [-7.642857142857146, 18.500000000000004, 0],
                              [-7.357142857142859, 18.500000000000004, 0],
                              [-7.214285714285717, 18.42857142857143, 0],
                              [-7.142857142857146, 18.35714285714286, 0],
                              [-7.071428571428573, 18.21428571428572, 0],
                              [-7.071428571428573, 18.071428571428577, 0],
                              [-7.142857142857146, 17.92857142857143, 0],
                              [-7.285714285714288, 17.71428571428572, 0],
                              [-8.000000000000002, 17.000000000000004, 0],
                              [-7.000000000000002, 17.000000000000004, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-5.285714285714288, 18.000000000000004, 0],
                              [-5.357142857142859, 17.92857142857143, 0],
                              [-5.285714285714288, 17.85714285714286, 0],
                              [-5.214285714285717, 17.92857142857143, 0],
                              [-5.285714285714288, 18.000000000000004, 0],
                            ],
                            [
                              null,
                              [-5.285714285714288, 17.142857142857146, 0],
                              [-5.357142857142859, 17.071428571428577, 0],
                              [-5.285714285714288, 17.000000000000004, 0],
                              [-5.214285714285717, 17.071428571428577, 0],
                              [-5.285714285714288, 17.142857142857146, 0],
                            ],
                            [null],
                            [null],
                          ],
                        },
                      ],
                    },
                    {
                      type: 'sketch',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'disjointAssembly',
                              content: [
                                {
                                  type: 'paths',
                                  tags: ['color/red'],
                                  paths: [
                                    [
                                      [
                                        16.000000000000004, 16.000000000000004,
                                        0,
                                      ],
                                      [
                                        -16.000000000000004, 16.000000000000004,
                                        0,
                                      ],
                                      [-16.000000000000007, -16, 0],
                                      [16, -16.000000000000007, 0],
                                    ],
                                  ],
                                },
                              ],
                            },
                            {
                              type: 'paths',
                              tags: ['color/red'],
                              paths: [
                                [
                                  null,
                                  [-15.642857142857146, 18.500000000000004, 0],
                                  [-14.857142857142861, 18.500000000000004, 0],
                                  [-15.285714285714288, 17.92857142857143, 0],
                                  [-15.071428571428575, 17.92857142857143, 0],
                                  [-14.928571428571432, 17.85714285714286, 0],
                                  [-14.857142857142861, 17.78571428571429, 0],
                                  [-14.785714285714288, 17.571428571428577, 0],
                                  [-14.785714285714288, 17.42857142857143, 0],
                                  [-14.857142857142861, 17.21428571428572, 0],
                                  [-15.000000000000004, 17.071428571428577, 0],
                                  [-15.214285714285717, 17.000000000000004, 0],
                                  [-15.428571428571432, 17.000000000000004, 0],
                                  [-15.642857142857146, 17.071428571428577, 0],
                                  [-15.714285714285717, 17.142857142857146, 0],
                                  [-15.78571428571429, 17.28571428571429, 0],
                                ],
                                [null],
                                [
                                  null,
                                  [-14.285714285714288, 18.142857142857146, 0],
                                  [-14.285714285714288, 18.21428571428572, 0],
                                  [-14.214285714285717, 18.35714285714286, 0],
                                  [-14.142857142857146, 18.42857142857143, 0],
                                  [-14.000000000000004, 18.500000000000004, 0],
                                  [-13.714285714285717, 18.500000000000004, 0],
                                  [-13.571428571428575, 18.42857142857143, 0],
                                  [-13.500000000000004, 18.35714285714286, 0],
                                  [-13.428571428571432, 18.21428571428572, 0],
                                  [-13.428571428571432, 18.071428571428577, 0],
                                  [-13.500000000000004, 17.92857142857143, 0],
                                  [-13.642857142857146, 17.71428571428572, 0],
                                  [-14.357142857142861, 17.000000000000004, 0],
                                  [-13.357142857142861, 17.000000000000004, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-11.785714285714288, 18.000000000000004, 0],
                                  [-11.000000000000004, 17.000000000000004, 0],
                                ],
                                [
                                  null,
                                  [-11.000000000000004, 18.000000000000004, 0],
                                  [-11.785714285714288, 17.000000000000004, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-9.285714285714288, 18.500000000000004, 0],
                                  [-8.500000000000004, 18.500000000000004, 0],
                                  [-8.92857142857143, 17.92857142857143, 0],
                                  [-8.714285714285717, 17.92857142857143, 0],
                                  [-8.571428571428573, 17.85714285714286, 0],
                                  [-8.500000000000004, 17.78571428571429, 0],
                                  [-8.42857142857143, 17.571428571428577, 0],
                                  [-8.42857142857143, 17.42857142857143, 0],
                                  [-8.500000000000004, 17.21428571428572, 0],
                                  [-8.642857142857146, 17.071428571428577, 0],
                                  [-8.85714285714286, 17.000000000000004, 0],
                                  [-9.071428571428573, 17.000000000000004, 0],
                                  [-9.285714285714288, 17.071428571428577, 0],
                                  [-9.35714285714286, 17.142857142857146, 0],
                                  [-9.42857142857143, 17.28571428571429, 0],
                                ],
                                [null],
                                [
                                  null,
                                  [-7.928571428571431, 18.142857142857146, 0],
                                  [-7.928571428571431, 18.21428571428572, 0],
                                  [-7.857142857142859, 18.35714285714286, 0],
                                  [-7.785714285714288, 18.42857142857143, 0],
                                  [-7.642857142857146, 18.500000000000004, 0],
                                  [-7.357142857142859, 18.500000000000004, 0],
                                  [-7.214285714285717, 18.42857142857143, 0],
                                  [-7.142857142857146, 18.35714285714286, 0],
                                  [-7.071428571428573, 18.21428571428572, 0],
                                  [-7.071428571428573, 18.071428571428577, 0],
                                  [-7.142857142857146, 17.92857142857143, 0],
                                  [-7.285714285714288, 17.71428571428572, 0],
                                  [-8.000000000000002, 17.000000000000004, 0],
                                  [-7.000000000000002, 17.000000000000004, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-5.285714285714288, 18.000000000000004, 0],
                                  [-5.357142857142859, 17.92857142857143, 0],
                                  [-5.285714285714288, 17.85714285714286, 0],
                                  [-5.214285714285717, 17.92857142857143, 0],
                                  [-5.285714285714288, 18.000000000000004, 0],
                                ],
                                [
                                  null,
                                  [-5.285714285714288, 17.142857142857146, 0],
                                  [-5.357142857142859, 17.071428571428577, 0],
                                  [-5.285714285714288, 17.000000000000004, 0],
                                  [-5.214285714285717, 17.071428571428577, 0],
                                  [-5.285714285714288, 17.142857142857146, 0],
                                ],
                                [null],
                                [null],
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          },
        },
      ];

      const run = async () => {
        const body = document.getElementsByTagName('body')[0];
        const notebookElement = await toDomElement(notebook);
        body.appendChild(notebookElement);
      };

      document.onreadystatechange = () => {
        if (document.readyState === 'complete') {
          run();
        }
      };
    </script>
  </body>
</html>
